### PR TITLE
Page create - Recycle Bin Check

### PR DESCRIPTION
**Pull request by leigh-pointer**
After Delete Page, Cant create page of same name #1645 issue. Added check and message if the page is in the recycle bin.

### CODE REVIEW

File: Oqtane.Client/Modules/Admin/Pages/Add.razor [hunk 1]
```diff
@@ -412,4 +418,9 @@
     {
         return !existingPages.Any(page => page.SiteId == siteId && page.Path == pagePath);
     }
+
+    private static bool PagePathIsDeleted(string pagePath, int siteId, List<Page> existingPages)
```
Comment (hishamco) [inline Oqtane.Client/Modules/Admin/Pages/Add.razor:422]: I suggest to move the LINQ query before the condition

File: Oqtane.Client/Modules/Admin/Pages/Add.razor [hunk 2]
```diff
@@ -322,6 +322,12 @@
                         }
                     }
 
+                    if(!PagePathIsDeleted(page.Path, page.SiteId, _pageList))
+                    {
+                        AddModuleMessage(string.Format(Localizer["Message.Page.Deleted"], _path), MessageType.Warning);
+                        return;
```
Comment (hishamco) [inline Oqtane.Client/Modules/Admin/Pages/Add.razor:328]: Add line break before 'return`

**Review by hishamco — CHANGES_REQUESTED** [review]
This check may be a workaround, the good solution is allow to create a page with the same name

**Comment by leigh-pointer** [comment]
> This check may be a workaround, the good solution is allow to create a page with the same name

you can't create a page with the same name because the database table Page has a UNIQUE NONCLUSTERED INDEX on it consisting of fields [SiteId],[Path],[UserId].

**Comment by hishamco** [comment]
You're right, I know that, but we may change this to make it an optimal solution

/cc @sbwalker

**Comment by sbwalker** [comment]
@leigh-pointer are you sure the logic below is correct? The condition "!PagePathIsDeleted()" seems to be saying "if the page is NOT deleted" then show the warning message - which does not seem accurate?

```
if(!PagePathIsDeleted(page.Path, page.SiteId, _pageList))
{
     AddModuleMessage(string.Format(Localizer["Message.Page.Deleted"], _path), MessageType.Warning);
    return;
}

**Pull request closed without merging**